Themis/ViewModels/Assessment/CodeEditorViewModel.swift

```diff
@@ -15,10 +15,27 @@ class CodeEditorViewModel: ObservableObject {
     @Published var selectedFile: Node?
     @Published var editorFontSize = CGFloat(14) // Default font size
     @Published var currentlySelecting: Bool = false
-    @Published var selectedLineNumber: Int?
+    @Published var selectedSection: NSRange? // First Range = selected Lines, Second Range = columns iff single line
     @Published var inlineHighlights: [String: [HighlightedRange]] = [:]
     @Published var showAddFeedback: Bool = false
 
+    var selectedSectionParsed: (NSRange, NSRange?)? {
+        if let selectedFile = selectedFile, let selectedSection = selectedSection, let lines = selectedFile.lines {
+            let fromLine = (lines.firstIndex(where: { $0.contains(selectedSection.location) }) ?? -1) + 1
+            let toLine = (lines.firstIndex(where: { $0.contains(selectedSection.location + selectedSection.length) }) ?? -1) + 1
```

Comment:
Shouldn't be unwrapped to -1 but to 0 as lines in file start with 1. Also if somebody should be able to get a line out of the range, app would crash due to out of bounds in column selection.

```suggestion
           let fromLine = (lines.firstIndex { $0.contains(selectedSection.location) } ?? 0) + 1
            let toLine = (lines.firstIndex { $0.contains(selectedSection.location + selectedSection.length) } ?? 0) + 1
```

Reply:
Changing this to zero makes it indistinguishable from feedbacks that are actually referenced to line 1. Therefore I kept it at -1 and added an extra check to prevent the out of bounds access. The displayed reference text for these feedbacks will now be an empty string, just like general feedbacks.